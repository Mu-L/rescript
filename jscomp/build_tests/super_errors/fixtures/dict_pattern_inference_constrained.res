let foo = dict =>
  switch dict {
  | @res.dictPattern {one: 1} =>
    let _: dict<string> = dict
    Js.log("one")
  | _ => Js.log("not one")
  }
