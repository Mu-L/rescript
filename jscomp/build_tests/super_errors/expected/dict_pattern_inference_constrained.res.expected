
  [1;33mWarning number 9[0m
  [36m/.../fixtures/dict_pattern_inference_constrained.res[0m:[2m3:22-29[0m

  1 [2mâ”‚[0m let foo = dict =>
  2 [2mâ”‚[0m   switch dict {
  [1;33m3[0m [2mâ”‚[0m   | @res.dictPattern [1;33m{one: 1}[0m =>
  4 [2mâ”‚[0m     let _: dict<string> = dict
  5 [2mâ”‚[0m     Js.log("one")

  the following labels are not bound in this record pattern: anyOtherField
Either bind these labels explicitly or add ', _' to the pattern.


  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/dict_pattern_inference_constrained.res[0m:[2m4:27-30[0m

  2 [2mâ”†[0m switch dict {
  3 [2mâ”†[0m | @res.dictPattern {one: 1} =>
  [1;31m4[0m [2mâ”†[0m   let _: dict<string> = [1;31mdict[0m
  5 [2mâ”†[0m   Js.log("one")
  6 [2mâ”†[0m | _ => Js.log("not one")

  This has type: [1;31mdict<int>[0m
  But it's expected to have type: [1;33mdict<string>[0m
  
  The incompatible parts:
    [1;31mint[0m vs [1;33mstring[0m
  
  You can convert [1;33mint[0m to [1;33mstring[0m with [1;33mBelt.Int.toString[0m.